x/gamm/keeper/pool_test.go

```diff
@@ -316,3 +320,38 @@ func (suite *KeeperTestSuite) TestGetPoolAndPoke() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestConvertToCFMMPool() {
+	ctrl := gomock.NewController(suite.T())
+
+	tests := map[string]struct {
+		pool        swaproutertypes.PoolI
+		expectError bool
+	}{
+		"cfmm pool": {
+			pool: mocks.NewMockCFMMPoolI(ctrl),
```

Comment:
For my learning, whats the benefit of doing this over running a KeeperTestSuite helper func to make a pool?

Reply:
Our helpers create a pool but don't return it. Instead, only return their id.

So we have to also call `GetPoolById()` to get the pool itself. It's more verbose.

Here, the goal is to simply test that given one pool implementation, the function passes. Given the other implementation, the function fails. So using mocks seemed to be the least verbose and simplest way to achieve that.